### ISSUE TITLE
GitHub App auth regression

### ISSUE THREAD

**Issue by joebowbeer** [argoproj/argo-cd#6196]
Checklist:

* [x] I've searched in the docs and FAQ for my answer: https://bit.ly/argocd-faq.
* [x] I've included steps to reproduce the bug.
* [x] I've pasted the output of `argocd version`.

**Describe the bug**

GitHub App auth via CLI and UI was broken in commit https://github.com/argoproj/argo-cd/commit/ac1808b1bee8f187e848fa314ba21afd549abd16

It looks like `GithubAppPrivateKeySecret` was accidentally left out when adding the CRLF/base64 validation.

@dpkirchner [added](https://github.com/team-rhino/argo-cd/commit/d1d9a542894a158c5f30daf1720318669a996c05) those properties to a local build and the rebuilt argo-cd instance can interact with the GH App auth'd repositories.

Half of the needed fix was applied in #6140

For more discussion see:
* #3086
* https://cloud-native.slack.com/archives/C01TSERG0KZ/p1620167623143200

**To Reproduce**

```bash
argocd repo add https://github.com/myorg/myrepo --github-app-id 1 --github-app-installation-id 2 --github-app-private-key-path myapp.private-key.pem
```

Similar issues with the UI.

**Expected behavior**

Adding a repo with GitHub App authentication should succeed.

**Version**

```shell
argocd: v2.0.1+33eaf11.dirty
  BuildDate: 2021-04-17T04:24:05Z
  GitCommit: 33eaf11e3abd8c761c726e815cbb4b6af7dcb030
  GitTreeState: dirty
  GoVersion: go1.16.3
  Compiler: gc
  Platform: darwin/amd64
argocd-server: v2.0.1+33eaf11
  BuildDate: 2021-04-15T22:31:51Z
  GitCommit: 33eaf11e3abd8c761c726e815cbb4b6af7dcb030
  GitTreeState: clean
  GoVersion: go1.16
  Compiler: gc
  Platform: linux/amd64
  Ksonnet Version: v0.13.1
  Kustomize Version: v3.9.4 2021-02-09T19:22:10Z
  Helm Version: v3.5.1+g32c2223
  Kubectl Version: v0.20.4
  Jsonnet Version: v0.17.0
```

**Comment by joebowbeer** [comment]
@jannfs @alexmt This issue was partially fixed in #6140 (at line 194 in repository.go). Both #6140 and #6197 will be needed in release-2.0. #6197 adds a unit test for #6140.

**Comment by jannfis** [comment]
Hey @joebowbeer, we'll cherry-pick #6140 and #6197 into the 2.0 branch. Thank you for the heads-up!
